spec/spec_helper.rb

```diff
@@ -82,16 +82,13 @@
       example.run
     end
   end
-end
 
-RSpec.shared_context 'with configured RubyLLM' do
-  before do
+  # Run once before all tests
+  config.before(:suite) do
     RubyLLM.configure do |config|
-      # NOTE: to ensure relevant models are pulled into your local server, do
-      #   bundle exec rake ollama:install_models_for_specs
       config.ollama_api_base_url = ENV.fetch('OLLAMA_API_BASE_URL', 'http://localhost:11434')
-
-      # FIXME: this clobbers the model list from models.json and results in all tests except for ollama's to fail
+      # needs to run when ONLY Ollama is configured and before any others are configured
+      # FIXME: RubyLLM.models.refresh!(provider: 'ollama') would be a cleaner solution
       RubyLLM.models.refresh!
```

Comment:
After yesterday's models logic change in main, doing `refresh!` when ONLY Ollama is configured fixes both the bogus outgoing calls and the vanishing models.json information problems.

This is a provisional commit; I'm sure there's a better way to organize this, but it exemplifies a solution without the need for `RubyLLM::Provider.disable_all_providers` (which was undone).